3djan/improve spec

jordig100 opened the pull request:
Improvements to the spec and making it ready to publish.

3dJan committed:
Running Formatter.py to update schema in markdown.

3dJan committed:
[implicit.xsd] abs(A) → 1 input
sqrt(A) → 1 input
clamp(A, min, max) → 3 inputs

3dJan committed:
Running Formatter.py to update schema in markdown.

jordig100 committed:
more schema fixes

jordig100 committed:
Update version and history

jordig100 committed:
Fixed TOC and appeendixes

3dJan committed:
[3MF Volumetric Extension] Rename arcsin, arccos, arctan, and arctan2 functions to asin, acos, atan, and atan2 respectively; update documentation accordingly

3dJan committed:
Merge branch '3djan/ImproveSpec' of ssh://ssh.github.com:443/3MFConsortium/spec_volumetric into 3dJan/ImproveSpec

3dJan committed:
[3MF Volumetric Extension Template] Update version to 1.0.0 and change status to Published

3dJan committed:
[3MF Volumetric Extension Template] Remove pre-release note from disclaimer section

3dJan committed:
Adding Jordi's changes also to "3MF Volumetric Extension Template.md" so it will survive running Formatter.py

3dJan committed:
[3MF Volumetric Extension] Rename arcsin, arccos, arctan, and arctan2 complex types to ASin, ACos, ATan, and ATan2 respectively

3dJan committed:
Updating images (arcsin to asin, ...)

3dJan committed:
Add Resources section for Volumetric and Implicit Extensions

3dJan committed:
Fix typos, improve formatting, and update references in 3MF Volumetric Extension documentation

- Corrected spelling errors and typos throughout the document.
- Improved the structure of the table of contents for clarity.
- Updated references to namespaces and XML elements for consistency.
- Clarified descriptions and notes related to volumetric data and functions.
- Adjusted formatting for better readability and adherence to documentation standards.
- Updated copyright year to 2025.

3dJan committed:
Remove image reference for PrivateExtensionFunction in documentation

3dJan commented on 3MF Volumetric Extension Template.md:
```diff
@@ -454,18 +470,18 @@ Element **\<volumedata>**
 To rationalize how this specification modifies the definition of geometry within a 3MF model, the concept of a "clipping surface" of a mesh with a `<volumedata>` element is introduced.
 The clipping surface is defined by the surface of the enclosing Shape. This implicitly takes into account any geometry defined by e.g. the beamlattices specification.
 
-This clipping surface trims any volumetric data defined therein. Any data outside the clipping surface MUST be ignored. The geometry that should be printed is defined by the interior of the clipping surface.
```
revert that change

3dJan commented on 3MF Volumetric Extension Template.md:
```diff
@@ -676,13 +692,23 @@ This specification is limited in scope. Three noteworthy limitations are:
+
+All new elements defined by the Implicit Extension also live under the core `<resources>` element.
+
+This extension adds `<i:implicitfunction>` resources in the implicit namespace, whose contents define a node graph of native operations and references.
```
add a note about how the order constrain is affected

3dJan committed:
[3MF Volumetric Extension Template] Introduce breaking change to core specification for resource ordering in volumetric extension

3dJan committed:
[3MF Volumetric Extension Template] Update table of contents to reflect breaking change and resource section reordering

3dJan committed:
[3MF Volumetric Extension] updating 3MF Volumetric Extension.md based on template

3dJan committed:
Reverting unintended change in wording

3dJan committed:
[Formatter.py] Add encoding specification for file operations to ensure proper reading and writing of files

3dJan committed:
[build.yml] Update workflow triggers to include workflow_dispatch

3dJan committed:
[mdToPDF.sh] Update PDF generation script to use safer rendering flags and add print stylesheet for improved layout

3dJan committed:
[print.css] Remove optional larger base font for print readability

3dJan committed:
[mdToPDF.sh] Adjust margin settings and remove global font-size injection for improved print layout

3dJan committed:
[print.css] Remove borders and shadows from README containers, adjust body font size to 10px, and ensure horizontal rules are hidden

3dJan committed:
[mdToPDF.sh] Update margin settings to align with displacement spec

3dJan committed:
[print.css] Enhance table typography for print with more compact font sizes and tighter padding

3dJan committed:
[mdToPDF.sh] Adjust top margin for improved layout consistency in print output
[print.css] Change body font size to 8px for better alignment with core specifications

3dJan committed:
[print.css] Update body font size to 11px for improved readability and override GitHub styles

3dJan committed:
[print.css] Adjust table font sizes for improved readability and consistency

3dJan committed:
[print.css] Adjust table layout and wrapping for improved readability and consistency

3dJan committed:
[print.css] Remove nowrap styles for short metadata columns and hint widths for table layout

3dJan committed:
Merge branch 'pdflayout' into 3djan/ImproveSpec

3dJan committed:
[3MF Volumetric Extension Template.md] Normalize Table of Contents indentation for sections 1.1 and 1.2

3dJan committed:
[3MF Volumetric Extension.md] Normalize Table of Contents indentation for sections 1.1 and 1.2

3dJan committed:
[implicit.xsd] Add FunctionGradient node to compute spatial gradients of referenced functions

3dJan committed:
[implicit.xsd] Fix typos and incorrect element/attribute names in documentation and examples

3dJan committed:
[3MF Volumetric Extension Template.md] Add functiongradient node documentation

- Add functiongradient entry to native nodes list
- Document functiongradient semantics: compute spatial gradient via central finite differences, return raw gradient, normalized vector and magnitude, and clamp step >= 1e-8
- Define inputs, outputs and attributes (scalaroutput, vectorinput, step with default 0.001)
- Include example XML usage for the functiongradient node

3dJan committed:
[implicit.xsd] Add SignedDistanceToBeamLattice node for evaluating signed distance to beam lattices and register beamlattice element

3dJan committed:
[3MF Volumetric Extension Template.md] Add beamlattice node documentation and register in native nodes list

3dJan committed:
[beamlattice] Add accuraterange attribute to node

Add optional "accuraterange" attribute (xs:double, default=0.0) to the SignedDistanceToBeamLattice node in implicit.xsd
Document semantics in schema documentation: non-negative distance band where returned signed distance MUST be accurate, outside which values may be approximate; require consumers to clamp negative accuraterange to 0.0
Update 3MF Volumetric Extension Template.md to include an Attributes table and example usage with accuraterange="2.5"

3dJan committed:
[functiongradient] Require scalar input "step" for finite-difference step; remove step attribute; define NaN/Inf/missing as undefined and enforce clamp >= 1e-8; update docs, schema and examples

3dJan committed:
[normalizedistance] Add normalizedistance node to specification and schema

- Add CT_NormalizeDistance to implicit.xsd and register <normalizedistance> element
- Document normalizedistance in 3MF Volumetric Extension Template.md and include example usage
- Define inputs/outputs and required attributes: resourceref "functionID", scalarref "step", attribute @scalaroutput, attribute @vectorinput
- Specify semantics: compute ∇f via central finite differences on the named vector input and output f(x)/max(|∇f(x)|, 1e-8)
- Require consumers to clamp effective step to >= 1e-8 and treat missing or non-finite step as undefined

3dJan committed:
[implicit.xsd, 3MF Volumetric Extension Template.md] Add beamlattice, functiongradient and normalizedistance nodes to schema and spec; register elements; document usage; change accuraterange type and fix schema refs and typos

3dJan committed:
[3MF Volumetric Extension Template.md, 3MF Volumetric Extension.md] Limiting TOC to chapter level

3dJan committed:
Merge branch '3djan/new_nodes' of ssh://ssh.github.com:443/3MFConsortium/spec_volumetric into 3djan/ImproveSpec

3dJan reviewed (approved):
Looks fine, PDF creation is now also fixed

jordig100 committed:
Set version back to draft

jordig100 committed:
Tentative publish date

3dJan committed:
Updating schema images

3dJan committed:
[3MF Volumetric Extension Template.md, 3MF Volumetric Extension.md] Update status to Published and updating publication date
